Contrasts for the rmANOVA

sjentsch opened the pull request:
- implemented contrasts for the rmANOVA
- added simple_1, and simple_k (simple contrasts vs. either first vs. last category) to all ANOVA, ANCOVA (and rmANOVA)
- checked the results vs. SPSS (correct / same apart from rounding errors at ~ second decimal)

sjentsch committed:
Implemented contrasts for the rmANOVA (incl. unit tests), added simple_k as contrast to AN(C)OVA

raviselker commented on R/ancova.b.R:
```diff
@@ -848,9 +853,16 @@ ancovaClass <- R6::R6Class(
             nLevels <- length(levels)
 
-            if (type == 'simple') {
+            if (type %in% c('simple_1', 'simple')) {
```
Why is this `simple_1` used as an alias? When using jamovi it still seems to be defined as just `simple` (which I think is also best to keep regarding backwards compatibility).

sjentsch replied:
I thought `simple_1` would make it clearer but I agree that it would make things rather more complicated when having an alias. I removed all occurrences of `simple_1`.

sjentsch replied:
Removed.

raviselker commented on R/anovarm.b.R:
```diff
@@ -894,6 +941,79 @@ anovaRMClass <- R6::R6Class(
         },
 
         #### Helper methods ----
+        .contrastLabels=function(levels, type) {
```
This function is almost the same as the one in `an(c)ova` no? Maybe it would be better to move this to the `utilsanova.R` file and then just use the same function in both ANOVAs

raviselker commented on R/anovarm.b.R:
```diff
@@ -1289,6 +1409,64 @@ calcUnivariateTests <- function(SSP, SSPE, P, df, error_df) {
+#' @param type Name / type of contrast (e.g., "simple_1")
+#' @return A function, used as input parameter `method` in emmeans::contrast
+#' @keywords internal
+defineContrasts <- function(type) {
```
This function returns a function right? So maybe name it something like `.getContrastFunction()` so that its more clear what it does.

raviselker reviewed (changes requested):
Great addition @sjentsch! I have a couple of remarks, but very happy that you implemented this!

sjentsch committed:
Changes requested by Ravi

sjentsch committed:
Making contrast labels translatable, deal with emmeans:::poly.emmc being limited to 6 degrees

jonathon-love commented:
your mods are including changes to the i18n submodule? i think you haven't quite ironed out your "git hygiene" just yet :P

raviselker commented on R/anovarm.h.R:
```diff
@@ -941,7 +941,10 @@ anovaRMBase <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
 #'   residuals
-#' @param contrasts in development
+#' @param contrasts a list of lists specifying the factor and type of contrast
+#'   to use, one of \code{'deviation'}, \code{'simple'} (or \code{'simple_1'}),
```
`simple_1` is still referenced here. As this is file is auto-generated I think you can just build the module again and this should be fixed.

raviselker commented on R/utilsanova.R:
```diff
@@ -12,3 +12,69 @@ setSingularityWarning = function(self) {
     )
 }
+
+#' Set a warning notice for singularity in the design matrix
```
Where does this tile come from? This is not what this function does right?

sjentsch replied:
Corrected.

raviselker reviewed (commented):
Some small things, but overall it looks good. As @jonathon-love mentioned, you have indeed also committed the i18n stuff so this needs to be fixed. Maybe jonathon can help you with this.

sjentsch committed:
Changed header files, removing simple_1

sjentsch committed:
Removed the last remains of simple_1

sjentsch commented:
Hei @raviselker, I removed all instances of simple_1 from the code, and corrected the description for the contrastLabels-function (). I promise to get better with my git hygiene. I had Jonathon's help with cleaning up some of the mess. Sorry for the additional work! Sebastian
